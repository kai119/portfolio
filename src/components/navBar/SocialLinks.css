.social-icon path {
  fill: none;
  stroke: #ecedee;
  stroke-width: 1px;
}
