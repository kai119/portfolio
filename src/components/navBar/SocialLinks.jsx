import {
  Box,
  Card,
  CardActionArea,
  CardContent,
  Icon,
  Typography,
  useTheme,
} from "@mui/material";
import React from "react";
import "./SocialLinks.css";
import EmailSvg from "./EmailIcon.svg";
import GithubSvg from "./GithubIcon.svg";
import LinkedinSvg from "./LinkedinIcon.svg";

function SocialLinks(props) {
  const theme = useTheme();

  const icons = [LinkedinSvg, GithubSvg, EmailSvg];

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Typography variant="subtitle2">Find me on</Typography>
      <Box
        sx={{
          margin: props.boxMargin,
          display: "flex",
          width: "100%",
          justifyContent: "space-evenly",
        }}
      >
        {icons.map((icon, index) => (
          <Card
            key={"icon-card-" + index}
            sx={{
              width: props.cardSize,
              height: props.cardSize,
              background: theme.palette.card.main,
              boxShadow: 2,
            }}
          >
            <CardActionArea
              sx={{
                display: "flex",
                alignItems: "center",
                justifyContent: "center",
              }}
            >
              <CardContent>
                <Icon>
                  <img
                    data-testid={"icon-img" + index}
                    alt="Social media icon"
                    src={icon}
                    height={props.iconSize}
                    width={props.iconSize}
                  />
                </Icon>
              </CardContent>
            </CardActionArea>
          </Card>
        ))}
      </Box>
    </Box>
  );
}

SocialLinks.defaultProps = {
  boxMargin: "0",
  cardSize: "30px",
  iconSize: "24px",
};

export default SocialLinks;
