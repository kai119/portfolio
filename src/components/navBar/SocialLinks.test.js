import React from "react";
import { render, screen } from "@testing-library/react";
import SocialLinks from "./SocialLinks";
import { ThemeProvider, createTheme } from "@mui/material";

it("renders with default props", () => {
  const theme = createTheme({ palette: { card: { main: "black" } } });
  const { container } = render(
    <ThemeProvider theme={theme}>
      <SocialLinks />
    </ThemeProvider>,
  );
  const { firstChild } = container;
  expect(firstChild).toMatchSnapshot();
  expect(screen.getByTestId("icon-img0").getAttribute("width")).toBe("24px");
  expect(screen.getByTestId("icon-img1").getAttribute("width")).toBe("24px");
  expect(screen.getByTestId("icon-img2").getAttribute("width")).toBe("24px");
});

it("correctly adjusts the width and height of the icon based on props", () => {
  const theme = createTheme({ palette: { card: { main: "black" } } });
  const { container } = render(
    <ThemeProvider theme={theme}>
      <SocialLinks iconSize="50px" />
    </ThemeProvider>,
  );
  const { firstChild } = container;
  expect(firstChild).toMatchSnapshot();
  expect(screen.getByTestId("icon-img0").getAttribute("width")).toBe("50px");
  expect(screen.getByTestId("icon-img1").getAttribute("width")).toBe("50px");
  expect(screen.getByTestId("icon-img2").getAttribute("width")).toBe("50px");
});
