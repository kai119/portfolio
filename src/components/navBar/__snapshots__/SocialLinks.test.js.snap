// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly adjusts the width and height of the icon based on props 1`] = `
<div
  class="MuiBox-root css-u6v8er"
>
  <h6
    class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    Find me on
  </h6>
  <div
    class="MuiBox-root css-1lm37yk"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qr7kfj-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-x7xklv-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            <img
              alt="Social media icon"
              data-testid="icon-img0"
              height="50px"
              src="LinkedinIcon.svg"
              width="50px"
            />
          </span>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qr7kfj-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-x7xklv-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            <img
              alt="Social media icon"
              data-testid="icon-img1"
              height="50px"
              src="GithubIcon.svg"
              width="50px"
            />
          </span>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qr7kfj-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-x7xklv-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            <img
              alt="Social media icon"
              data-testid="icon-img2"
              height="50px"
              src="EmailIcon.svg"
              width="50px"
            />
          </span>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`renders with default props 1`] = `
<div
  class="MuiBox-root css-u6v8er"
>
  <h6
    class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    Find me on
  </h6>
  <div
    class="MuiBox-root css-1lm37yk"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qr7kfj-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-x7xklv-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            <img
              alt="Social media icon"
              data-testid="icon-img0"
              height="24px"
              src="LinkedinIcon.svg"
              width="24px"
            />
          </span>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qr7kfj-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-x7xklv-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            <img
              alt="Social media icon"
              data-testid="icon-img1"
              height="24px"
              src="GithubIcon.svg"
              width="24px"
            />
          </span>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qr7kfj-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-x7xklv-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          >
            <img
              alt="Social media icon"
              data-testid="icon-img2"
              height="24px"
              src="EmailIcon.svg"
              width="24px"
            />
          </span>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
